src/hooks/useOrgHooks.js

```diff
@@ -179,39 +180,46 @@ export function useTokenBalances(account, token, timer = 3000) {
     let cancelled = false
     let timeoutId
 
-    const fetchAccountStakeBalance = async () => {
-      try {
-        let contractNewBalance = new BigNumber(-1)
-        if (account) {
-          contractNewBalance = await tokenContract.balanceOf(account)
-        }
+    const pollAccountBalance = async () => {
+      timeoutId = setTimeout(async () => {
+        try {
+          let contractNewBalance = new BigNumber(-1)
+          if (account) {
+            contractNewBalance = await tokenContract.balanceOf(account)
+          }
 
-        const contractTotalSupply = await tokenContract.totalSupply()
+          const contractTotalSupply = await tokenContract.totalSupply()
 
-        if (!cancelled) {
-          // Contract value is bn.js so we need to convert it to bignumber.js
-          const newBalance = new BigNumber(contractNewBalance.toString())
-          const newTotalSupply = new BigNumber(contractTotalSupply.toString())
-
-          if (
-            !newTotalSupply.eq(balances.totalSupply) ||
-            !newBalance.eq(balances.balance)
-          ) {
-            setBalances({ balance: newBalance, totalSupply: newTotalSupply })
+          if (!cancelled) {
+            // Contract value is bn.js so we need to convert it to bignumber.js
+            const newBalance = new BigNumber(contractNewBalance.toString())
+            const newTotalSupply = new BigNumber(contractTotalSupply.toString())
+
+            if (
+              !newTotalSupply.eq(balances.totalSupply) ||
+              !newBalance.eq(balances.balance)
+            ) {
+              setBalances({ balance: newBalance, totalSupply: newTotalSupply })
+            }
           }
+        } catch (err) {
+          console.error(`Error fetching balance: ${err} retrying...`)
         }
-      } catch (err) {
-        console.error(`Error fetching balance: ${err} retrying...`)
-      }
+        if (!cancelled) {
+          clearTimeout(timeoutId)
+          controlledTimeout.current = timeout
+          pollAccountBalance()
+        }
+      }, controlledTimeout.current)
```

Comment:
There's a cleaner version of this that we may want to use instead.
In this way we don't need to controlled timeout ref

```
    const pollAccountBalance = async () => {
      try {
        let contractNewBalance = new BigNumber(-1)
        if (account) {
          contractNewBalance = await tokenContract.balanceOf(account)
        }

        const contractTotalSupply = await tokenContract.totalSupply()

        if (!cancelled) {
          // Contract value is bn.js so we need to convert it to bignumber.js
          const newBalance = new BigNumber(contractNewBalance.toString())
          const newTotalSupply = new BigNumber(contractTotalSupply.toString())

          if (
            !newTotalSupply.eq(balances.totalSupply) ||
            !newBalance.eq(balances.balance)
          ) {
            setBalances({ balance: newBalance, totalSupply: newTotalSupply })
          }
        }
      } catch (err) {
        console.error(`Error fetching balance: ${err} retrying...`)
      }
      if (!cancelled) {
        timeoutId = setTimeout(pollAccountBalance, timeout)
      }
    }
```

Reply:
ah i think this is the same and we also have that logic on the `useVaultBalance`